File: C++/代码随想录/String/28. Find the Index of the First Occurrence in a String.cpp
// 28. Find the Index of the First Occurrence in a String
// https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/

#include <iostream>

int firstOccurrence(std::string haystack, std::string needle) {
    return haystack.find(needle);
}

int main() {
    std::string haystack = "hello world";
    std::string needle = "world";
    // std::string needle = "qq";
    
    int result = firstOccurrence(haystack, needle);
    
    std::cout << "Index of first occurrence: " << result << std::endl;
    
    return 0;
}

File: C++/代码随想录/String/28. Find the Index of the First Occurrence in a String.py
# 不要用这个，用cpp
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if not needle:
            return 0
        for i in range(len(haystack) - len(needle) + 1):
            if haystack[i:i+len(needle)] == needle:
                return i
        return -1

sol = Solution()

# 测试用例
print(sol.strStr("hello", "ll"))  # 输出: 2
print(sol.strStr("aaaaa", "bba")) # 输出: -1
print(sol.strStr("", ""))         # 输出: 0
